packages/editor/src/components/post-author/index.js

```diff
@@ -97,6 +115,29 @@ function PostAuthor() {
 		return null;
 	}
 
+	if ( ! isCombobox ) {
+		const selectId = 'post-author-selector-' + instanceId;
+
+		return (
+			<PostAuthorCheck>
+				<label htmlFor={ selectId }>{ __( 'Author' ) }</label>
+				<select
+					id={ selectId }
+					defaultValue={ postAuthor.id }
+					onBlur={ handleSelect }
+					className="editor-post-author__select"
+				>
+					{ authors &&
+						authors.map( ( author ) => (
+							<option key={ author.id } value={ author.id }>
+								{ decodeEntities( author.name ) }
+							</option>
+						) ) }
+				</select>
+			</PostAuthorCheck>
+		);
+	}
+
```

Comment:
I'm not sure I like the fact that both implementations are inside the same component, what if it's instead  two subcomponents and the parent component just switches from one to another depending on the result of `getEntityRecords`?

Reply:
I can give this a try, it might make the code easier to understand.